### PR TITLE
EWLJ-323: Scroll vertically inside magnified images

### PR DESCRIPTION
**Pull request by shaal**
## Ticket(s)

Please do not submit a Pull Request without a relevant JIRA ticket or Github issue! If you are creating a new pattern or feature, create an issue describing the need for this feature in Github **first** and then link to it below.

**Jira Ticket**

- [EWLJ-323: Need figures and images to magnify when magnified](https://issues.ama-assn.org/browse/EWLJ-323)

## Description

Removed CSS rules that limited the image from being displayed full size.

## To Test
- [ ] Switch your JOE SG2 branch to `EWLJ-323-magnify-full-screen-images`
- [ ] Run gulp serve
- [ ] Enable local SG2 in your D8 project
- [ ] In another terminal run `drush @joe.local cr`
- [ ] Go to http://ama-joe.local/article/why-we-need-music-player-every-patient-room/2019-03
- [ ] Click on the magnifying glass in the top-right corner of an image
- [ ] You should see the image displayed on the whole screen, with `X` icon on the top-right corner of the image
- [ ] Confirm that you can scroll down if the image is hight than your screen's resolution
- [ ] Clicking anywhere on the image will close the magnified display and go back to the article page

## Relevant Screenshots/GIFs

Use something like [Skitch](https://evernote.com/skitch/) or [GIPHY Capture](https://giphy.com/apps/giphycapture) to capture images/gifs to demonstrate behaviors.
images-magnified-in-full-screen

### CODE REVIEW

**Comment by designerneil** [comment]
The scrolling does not scroll the image 
Screen Shot 2019-04-16 at 4 29 33 PM

**Comment by shaal** [comment]
@designerneil When I test the patch again, after clicking on magnifying the image, I do see the scroll-bar and can scroll down. See the screenshot:
image

**Comment by shaal** [comment]
@designerneil I think I covered all the edge cases and various browsers where the problem occurred.
With this last commit, this should all be resolved and work as planned.

**Comment by designerneil** [comment]
@shaal when the magnifier is clicked the image does indeed get larger and I'm able to scroll. But when I close the image the page has scrolled to the top and not maintained my position before I clicked the image. I expect the page scroll position to stay when the magnify is clicked.

**Comment by designerneil** [comment]
@shaal the close X icon doesn't show up in IE11
Screen Shot 2019-04-23 at 12 52 54 PM

**Comment by shaal** [comment]
@designerneil I updated the code to be compatible with IE11, and it is now 'remembers' the original scroll position when closing the magnified image.

File: styleguide/source/assets/js/lightbox.js [hunk 1]
```diff
@@ -7,25 +7,42 @@
...
+        $(this).parents('.joe__inline-image').toggleClass('is-zoomed');
+        // While image is zoomed-in, save the scroll position and freeze scrolling.
+        if ($('.layout-container').hasClass('freeze-scrolling')) {
+          $('.layout-container').removeClass('freeze-scrolling');
+          // Restore scroll position before image was magnified.
+          $(document).scrollTop($savedScrollPosition);
+          $savedScrollPosition = 0;
+        } else {
+          // Save scroll position during 'freeze-scrolling'.
+          $savedScrollPosition = $(document).scrollTop();
```
Comment (designerneil) [inline styleguide/source/assets/js/lightbox.js:30]:
```suggestion
          $savedScrollPosition = $(window).scrollTop();
```

Comment (designerneil) [reply styleguide/source/assets/js/lightbox.js:30]: If it's on document the page still scrolls a little bit when the user scroll within the image

**Review by designerneil — CHANGES_REQUESTED** [review]
just change the document.scroll to window

**Comment by shaal** [comment]
I didn't try the suggestion yet but I'm pretty sure they do the same, and
if it's what I saw yesterday the little 'scroll' that you so see is
probably the effect of header growing to its original height.
When I tested it, the actual scroll position was exactly how it was when
entering the magnified image mode.

**Review by designerneil — APPROVED** [review]

**Comment by designerneil** [comment]
@shaal I'm closing this PR because it's been replaced by an alternative path the client has chosen. The story for alternative path is https://issues.ama-assn.org/browse/EWLJ-406

**Comment by designerneil** [comment]
@shaal @cngoody This PR has been merged in and works as described. You are able to scroll within enlarged images.

**Pull request merged**